crates/noirc_frontend/src/monomorphisation/mod.rs

```diff
@@ -387,7 +387,7 @@ impl Monomorphiser {
             definitions.push(new_expr);
         }
 
-        ast::Expression::Block(definitions)
+        ast::Expression::Tuple(definitions)
```

Comment:
Why was this changed to a tuple? The intention here was to extract each field of the tuple into a separate let binding in the same block, so:
```rs
let (a, b, c) = (1, 2, 3);
```
becomes
```rs
{
    let _ = (1, 2, 3);
    let a = _.0;
    let b = _.1;
    let c = _.2;
}
```
No need to return a tuple

Reply:
At some point I was experimenting with location in block of expressions and switching to a Tuple allowed me not to handle the case here, and it did not break anything. I decided to keep the change to get your feedback.
Isn't returning a tuple the correct thing to do here?

Reply:
It isn't correct in general since this isn't meant to be returning a tuple. This function is called when compiling let bindings that have a tuple pattern as the lhs, like the first example I gave above. So it is meant to expand it into 3+1 separate let bindings instead so later passes don't have to worry about unpacking/repacking values so much. This allows them to have "1 let binding = 1 DefinitionId" which isn't true if the let binding defines multiple variables in a tuple/struct pattern.

Now, since let bindings only return unit values which are ignored, it is technically fine if they return a tuple instead as it should be ignored as well. However, this means there is no reason to construct the tuple to begin with.

If you want to add a location to block patterns, I'd say it is fine to add a location to let bindings in general and grab that one.

Reply:
I have reverted it